dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/map/GeoData.java

```diff
@@ -0,0 +1,72 @@
+package org.dashbuilder.renderer.c3.client.charts.map;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Map.Entry;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.dashbuilder.renderer.c3.client.exports.NativeLibraryResources;
+
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONParser;
+import com.google.gwt.json.client.JSONValue;
+
+@ApplicationScoped
+public class GeoData {
+    
+    private JSONArray countries;
+
+    @PostConstruct
+    public void loadGeoJson() {
+        String jsonString = NativeLibraryResources.INSTANCE.countriesgeojson().getText();
+        countries = JSONParser.parseStrict(jsonString).isObject().get("features").isArray();
+    }
+    
+    public JSONArray countries() {
+        return countries;
+    }
+    
+    public  String countryName(JSONValue value) {
+        return value.isObject().get("properties").isObject().get("name").toString().replaceAll("\"", "");
+    }
+    
+    public String countryID(JSONValue value) {
+        return value.isObject().get("id").toString().replaceAll("\"", "");
+    }
+    
+    public Optional<JSONValue> findPathByCountryIdOrName(String idOrName) {
+        JSONValue result = null;
+        for (int i = 0; i < countries.size(); i++) {
+            JSONValue value = countries.get(i);
+            if (idOrName.equalsIgnoreCase(countryID(value)) || 
```

Comment:
This in my opinion represents usability issue of Map displayers: 
I find it quite hard to know what country names / country codes users should use in their data sets to associate values with individual countries.
Are those the things like `"id":"PAK","properties":{"name":"Pakistan"}`, from countries.geo.json?
Is that documented somewhere?

Reply:
The country ID should be country code based on this ISO_3166.

https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3

The JSON that is used should follow GeoJson standard, which is compatible with d3-geoprojection https://tools.ietf.org/html/rfc7946

As I said, indeed I will improve the access to GeoJSON. I didn't before because "properties" is a free map for properties, in the GeoJSON we are using I can retrieve the name for it, but it may not be the case for other GeoJSON, so I can't rely on a hardcoded name property. 

--
I do agree that the maps usage is obscure, but I do plan to document it, create tutorials. IMO also we should expand support for maps in future, like set center for focus in certain region. I was planning to add navigation buttons to this displayer (move and zoom), like the ones from google maps, but I realized I would never finish, ideas will never stop coming :)